tapdb/sqlc/queries/assets.sql

```diff
@@ -776,6 +776,86 @@ SELECT key_family, key_index
 FROM internal_keys
 WHERE raw_key = $1;
 
+-- name: UpsertTapscriptTreeRootHash :one
+INSERT INTO tapscript_roots (
+    root_hash, branch_only
+) VALUES (
+    $1, $2
+) ON CONFLICT (root_hash)
+    -- This is a NOP, the root_hash is the unique field that caused the
+    -- conflict. The tree should be deleted before switching between branch and
+    -- leaf storage.
+    DO UPDATE SET root_hash = EXCLUDED.root_hash
+RETURNING root_id;
+
+-- name: UpsertTapscriptTreeNode :one
+INSERT INTO tapscript_nodes (
+    raw_node
+) VALUES (
+    $1
+) ON CONFLICT (raw_node)
+    -- This is a NOP, raw_node is the unique field that caused the conflict.
+    DO UPDATE SET raw_node = EXCLUDED.raw_node
+RETURNING node_id;
+
+-- name: UpsertTapscriptTreeEdge :one
+INSERT INTO tapscript_edges (
+    root_hash_id, node_index, raw_node_id
+) VALUES (
+    $1, $2, $3
+) ON CONFLICT (root_hash_id, node_index, raw_node_id)
+    -- This is a NOP, root_hash_id, node_index, and raw_node_id are the unique
+    -- fields that caused the conflict.
+    DO UPDATE SET root_hash_id = EXCLUDED.root_hash_id,
+    node_index = EXCLUDED.node_index, raw_node_id = EXCLUDED.raw_node_id
+RETURNING edge_id;
+
+-- name: FetchTapscriptTreeRoots :many
+SELECT *
+FROM tapscript_roots;
+
+-- name: FetchTapscriptTree :many
+WITH tree_info AS (
+    -- We use this CTE to fetch all edges that link the given tapscript tree
+    -- root hash to child nodes. Each edge also contains the index of the child
+    -- node in the tapscript tree.
+    SELECT tapscript_roots.root_id, tapscript_roots.root_hash,
```

Comment:
nit: since we only use the `branch_only`, `raw_node_id` and `node_index` in the actual outer select, do we need to select the other fields at all?